test_runner/fixtures/compare_fixtures.py

```diff
@@ -340,23 +340,27 @@ def neon_with_baseline(request: FixtureRequest) -> PgCompare:
 
 
 @pytest.fixture(scope="function", autouse=True)
-def sync_after_each_test():
-    # The fixture calls `sync(2)` after each test if `SYNC_AFTER_EACH_TEST` env var is `true`
+def sync_between_tests():
+    # The fixture calls `sync(2)` after each test if `SYNC_BETWEEN_TESTS` env var is `true`
     #
-    # In CI, `SYNC_AFTER_EACH_TEST` is set to `true` only for benchmarks (`test_runner/performance`)
+    # In CI, `SYNC_BETWEEN_TESTS` is set to `true` only for benchmarks (`test_runner/performance`)
     # that are run on self-hosted runners because some of these tests are pretty write-heavy
     # and create issues to start the processes within 10s
-    key = "SYNC_AFTER_EACH_TEST"
+    key = "SYNC_BETWEEN_TESTS"
     enabled = os.environ.get(key) == "true"
 
-    yield
+    if enabled:
```

Comment:
Optional: Maybe we can keep the circuit breaker instead of 2 if's?

```
if not enabled:
  return 
.
.
.
yield
.
.
.
```

Reply:
I think if we return before `yield` then the test won't run

Reply:
We can `return` from the fixture, that's ok (the test will be run, just checked it locally)

Reply:
Oh, that's spooky.  My mental model of python generators is apparently broken